devices/utilities.c

```diff
@@ -110,18 +95,12 @@ int16_t *get_reg_ptr(Emulator *emu, uint8_t reg)
     Cpu *cpu = emu->cpu;
 
     static int16_t r2 = 0;
-    
+
     switch (reg)
-    {    
+    {
         case 0x0: return (int16_t *) &cpu->pc;
         case 0x1: return (int16_t *) &cpu->sp;
-
-        case 0x2:
-        {
-            r2 = sr_to_value(emu);
-            return &r2;
-        }
-            
+        case 0x2: return (int16_t *)(void*)&cpu->sr;
```

Comment:
cast through `void*`? it usually means something strange...

Reply:
Yes, casting a structure with bitfields.

Reply:
so it's a UB, it can vary between gcc versions...

Reply:
SR handling was rewritten